src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Jint.Native;
+using Nethermind.Blockchain.Find;
+using Nethermind.Core;
+using Nethermind.Int256;
+using Nethermind.TxPool;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public class GasPrice
```

Comment:
name of this class does not represent what the class does.
Is this class a gas price? not
Does this class calculate the gas price to be returned by JSON RPC? yes

name should be more like MinGasPriceEstimator, GasPriceEstimator, GasPriceOracle

Reply:
Ideally such a class would be behind an interface IGasPriceOracle
(gas price oracle is some magical provider of the current gas price)
Implementation can be internal, or based on GasNow.org etc